go-controller/pkg/node/gateway_iptables.go

```diff
@@ -129,33 +129,18 @@ func getNodePortIPTRules(svcPort kapi.ServicePort, nodeIP *net.IPNet, targetIP s
 	} else {
 		protocol = iptables.ProtocolIPv4
 	}
-	var natArgs, filterArgs []string
-	if nodeIP != nil {
-		natArgs = []string{
-			"-p", string(svcPort.Protocol),
-			"-d", nodeIP.IP.String(),
-			"--dport", fmt.Sprintf("%d", svcPort.NodePort),
-			"-j", "DNAT",
-			"--to-destination", util.JoinHostPortInt32(targetIP, targetPort),
-		}
-		filterArgs = []string{
-			"-p", string(svcPort.Protocol),
-			"-d", nodeIP.IP.String(),
-			"--dport", fmt.Sprintf("%d", svcPort.NodePort),
-			"-j", "ACCEPT",
-		}
-	} else {
-		natArgs = []string{
-			"-p", string(svcPort.Protocol),
-			"--dport", fmt.Sprintf("%d", svcPort.NodePort),
-			"-j", "DNAT",
-			"--to-destination", util.JoinHostPortInt32(targetIP, targetPort),
-		}
-		filterArgs = []string{
-			"-p", string(svcPort.Protocol),
-			"--dport", fmt.Sprintf("%d", svcPort.NodePort),
-			"-j", "ACCEPT",
-		}
+	natArgs := []string{
+		"-p", string(svcPort.Protocol),
+		"-d", nodeIP.IP.String(),
```

Comment:
just using ```"--dst-type", "LOCAL",``` should be enough, no need configure the IPs

Reply:
I think Antonio is right, if we use --dst-type it will cover node port to all self-hosted addresses right?

Reply:
Looking at the k8s service doc page I see:
```
If the --nodeport-addresses flag in kube-proxy is set, would be filtered NodeIP(s).)
```
So looks like it is configurable in kube-proxy whether or not to specify to filter based on specific node IP. I wonder if @danwinship has any thoughts on this...

Reply:
you can specify in which ips of the hosts you want to expose the nodeports

Reply:
ACK, I will just change to `--dst-type LOCAL` and deal with any additional features, such as filtering node IPs for node port services, in a later PR, if we'll ever find the need to do so.

Reply:
yeah, kube-proxy offers the option of only exporting node ports on a single node IP, but that's not a feature of the Service API, it's a feature of kube-proxy, and we don't have to copy it

Reply:
ok then I guess we are good with applying to all IPs on the node then